dev/TEAL.md

```diff
@@ -134,45 +134,45 @@ Some of these have immediate data in the byte or bytes after the opcode.
 
 **Transaction Fields**
 
-| Index | Name | Type |
-| --- | --- | --- |
-| 0 | Sender | []byte |
-| 1 | Fee | uint64 |
-| 2 | FirstValid | uint64 |
-| 3 | FirstValidTime | uint64 |
-| 4 | LastValid | uint64 |
-| 5 | Note | []byte |
-| 6 | Lease | []byte |
-| 7 | Receiver | []byte |
-| 8 | Amount | uint64 |
-| 9 | CloseRemainderTo | []byte |
-| 10 | VotePK | []byte |
-| 11 | SelectionPK | []byte |
-| 12 | VoteFirst | uint64 |
-| 13 | VoteLast | uint64 |
-| 14 | VoteKeyDilution | uint64 |
-| 15 | Type | []byte |
-| 16 | TypeEnum | uint64 |
-| 17 | XferAsset | uint64 |
-| 18 | AssetAmount | uint64 |
-| 19 | AssetSender | []byte |
-| 20 | AssetReceiver | []byte |
-| 21 | AssetCloseTo | []byte |
-| 22 | GroupIndex | uint64 |
-| 23 | TxID | []byte |
+| Index | Name | Type | Notes |
+| --- | --- | --- | --- |
+| 0 | Sender | []byte | 32 byte address |
+| 1 | Fee | uint64 | micro-Algos |
+| 2 | FirstValid | uint64 | round number |
+| 3 | FirstValidTime | uint64 | Seconds since 1970-01-01 00:00:00 UTC of block at FirstValid-1 |
+| 4 | LastValid | uint64 | round number |
+| 5 | Note | []byte |  |
+| 6 | Lease | []byte |  |
+| 7 | Receiver | []byte | 32 byte address |
+| 8 | Amount | uint64 | micro-Algos |
+| 9 | CloseRemainderTo | []byte | 32 byte address |
+| 10 | VotePK | []byte | 32 byte address |
+| 11 | SelectionPK | []byte | 32 byte address |
+| 12 | VoteFirst | uint64 |  |
+| 13 | VoteLast | uint64 |  |
+| 14 | VoteKeyDilution | uint64 |  |
+| 15 | Type | []byte |  |
+| 16 | TypeEnum | uint64 | See table below |
+| 17 | XferAsset | uint64 | Asset ID |
+| 18 | AssetAmount | uint64 | value in Asset's units |
+| 19 | AssetSender | []byte | 32 byte address. Causes clawback of all value of asset from AssetSender if Sender is the Clawback address of the asset. |
+| 20 | AssetReceiver | []byte | 32 byte address |
+| 21 | AssetCloseTo | []byte | 32 byte address |
+| 22 | GroupIndex | uint64 | Position of this transaction within an atomic transaction group. A stand-alone transaction is implicitly element 0 in a group of 1. |
+| 23 | TxID | []byte | The computed ID for this transaction. 32 bytes. |
 
 
 Additional details in the [opcodes document](TEAL_opcodes.md#txn) on the `txn` op.
 
 **Global Fields**
```

Comment:
Can you add:
"Global fields are fields that are common to all the transactions in the group. In particular it includes consensus parameters."